src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ConcreteBackfillerBase.java

```diff
@@ -132,11 +157,12 @@ protected boolean workingUpward(ItemStack aStack, int xDrill, int yDrill, int zD
 
     private boolean isRefillableBlock(int aX, int aY, int aZ) {
         IGregTechTileEntity aBaseTile = getBaseMetaTileEntity();
-        if (!aBaseTile.getBlock(aX, aY, aZ)
-            .isAir(aBaseTile.getWorld(), aX, aY, aZ) || aBaseTile.getBlock(aX, aY, aZ)
-                .getMaterial()
-                .isSolid())
-            return false;
+        Block aBlock = getBaseMetaTileEntity().getBlock(aX, aY, aZ);
+        if (mLiquidEnabled) {
+            if (!aBaseTile.getBlock(aX, aY, aZ)
+                .isAir(aBaseTile.getWorld(), aX, aY, aZ) && !isFluid(aBlock)) return false;
+        } else if (!aBaseTile.getBlock(aX, aY, aZ)
+            .isAir(aBaseTile.getWorld(), aX, aY, aZ)) return false;
```

Comment:
I reorganized conditions to remove unnecessary repeated method calls and restore original logic:
```suggestion
        Block aBlock = aBaseTile.getBlock(aX, aY, aZ);
        if (!aBlock.isAir(aBaseTile.getWorld(), aX, aY, aZ)) {
            if (mLiquidEnabled) {
                if (!isFluid(aBlock)) {
                    return false;
                }
            } else {
                // remove this branch if liquid mode is not exclusive.
                return false;
            }
        }
        if (aBlock.getMaterial().isSolid()) {
            return false;
        }
```

Am I correctly understanding that liquid mode should be exclusive? I. e. machine replaces only air-like blocks (liquid mode=off) or only fluids (liquid mode=on)?
If machine should remove both types of blocks when liquid mode is enabled,  then remove branch with comment.

Reply:
> I reorganized conditions to remove unnecessary repeated method calls and restore original logic:
> 
> Am I correctly understanding that liquid mode should be exclusive? I. e. machine replaces only air-like blocks (liquid mode=off) or only fluids (liquid mode=on)? If machine should remove both types of blocks when liquid mode is enabled, then remove branch with comment.

When liquid mode is turned on, both air and liquid must be replaced

Reply:
If so, simplify your code using snippet above ***without*** ```else``` branch.

Reply:
> If so, simplify your code using snippet above _**without**_ `else` branch.

Changed it now correctly?